source/common/tls/utility.cc

```diff
@@ -226,6 +226,10 @@ std::string Utility::generalNameAsString(const GENERAL_NAME* general_name) {
     }
     break;
   }
+  case GEN_OTHERNAME:
+    str = general_name->d.otherName->value->value.asn1_string;
```

Comment:
Can any of the pointers in this chain be null?

Reply:
general_name->d.otherName should be present given the type and also the matchOID that occurs before this function call from san_matcher rules it out. value.asn1_string points to the address irrespective of type, if value exists. otherName->value is a possible  nullptr, although openssl/other alternatives disallows setting oid without value. Added a sanity check here, in any case

Reply:
While digging into this, I realized we need to handle a few types differently - https://github.com/openssl/openssl/blob/master/crypto/asn1/asn1_gen.c#L601C5-L601C6 . I've updated the code to handle these different types while fitting within our string matching paradigm. Please let me know what you think